test/controllers/concerns/child_resource_test.rb

```diff
@@ -0,0 +1,130 @@
+require 'test_helper'
+
+class Son
+  extend ActiveModel::Naming
+  attr_accessor :mom, :id
+
+  def destroyed?
+    true
+  end
+
+  def persisted?
+    false
+  end
+
+  def to_model
+    self
+  end
+
+  def to_param
+    id
+  end
+
+  def self.where(*_args)
+    @@singleton ||= new.tap { |s| s.mom = Mom.new }
+  end
+
+  def build
+    self
+  end
+
+  def initialize
+    self.id = 1
+  end
+end
+
+class Mom
+  extend ActiveModel::Naming
+  attr_accessor :son, :id
+
+  def persisted?
+    false
+  end
+
+  def to_model
+    self
+  end
+
+  def to_param
+    id
+  end
+
+  def self.find(*_args)
+    @@singleton ||= new.tap { |m| m.son = Son.new }
+  end
+
+  def initialize
+    self.id = 1
+  end
+end
+
+class Api::SonsController < ActionController::Base
+  include HalApi::Controller
+  include ChildResource
+
+  child_resource parent: 'mom', child: 'son'
+
+  def authorize(_r)
+    true
+  end
+end
+
+class Api::SonRepresenter < Api::BaseRepresenter
+  property :id
+
+  def self_url(represented)
+    api_mom_son_path(represented.mom)
+  end
+end
+
+def define_child_routes
+  Rails.application.routes.draw do
+    namespace :api do
+      resources :moms do
+        resource :son
+      end
+    end
+  end
+end
+
+describe Api::SonsController do
```

Comment:
I know, srsly, I do have some test models I use for this sort of thing, but they weren't quite right:
https://github.com/PRX/cms.prx.org/blob/master/test/test_models.rb

Maybe I should add the test setup to `test_models` instead of in this test?